_posts/2016/2016-09-08-kubernetes-preparing-nodes.md

```diff
@@ -0,0 +1,259 @@
+root@kubernetes:/etc/ssl/etcd$ ls -la
+-rw-r--r--  1 root root 4711 Jun  33 08:15 ca.crt
+-rw-r--r--  1 root root 4712 Jun  33 08:15 etcd-client.crt
+-rw-------  1 root root 4713 Jun  33 08:15 etcd-client.key
+{% endhighlight %}
+
+### Installation of the daemon
+
+For the various settings an extra file __options.env__ is used.
+Besides all *Etcd* nodes it contains the certificates.
+Finally the *Systemd* service is created.
+
+{% highlight bash %}
+root@kubernetes:~$ cd /tmp
+root@kubernetes:/tmp$ mkdir -p /opt/flanneld
+root@kubernetes:/tmp$ curl -L -O https://github.com/coreos/flannel/releases/download/v0.5.5/flannel-0.5.5-linux-amd64.tar.gz
+root@kubernetes:/tmp$ tar -xzf flannel-0.5.5-linux-amd64.tar.gz
+root@kubernetes:/tmp$ cp flannel-0.5.5/flanneld /usr/local/bin/flanneld
+root@kubernetes:/tmp$ cp flannel-0.5.5/mk-docker-opts.sh /opt/flanneld/mk-docker-opts.sh
+root@kubernetes:/tmp$ chmod 755 /usr/local/bin/flanneld
+root@kubernetes:/tmp$ chmod 755 /opt/flanneld/mk-docker-opts.sh
+root@kubernetes:/tmp$ mkdir -p /etc/flanneld
+root@kubernetes:/tmp$ export PRIMARY_HOST_IP=192.168.1.[1|2] or 192.168.2.[1|2|3]
+root@kubernetes:/tmp$ cat > /etc/flanneld/options.env << EOF
+FLANNELD_ETCD_ENDPOINTS=https://192.168.0.1:2379,https://192.168.0.2:2379,https://192.168.0.3:2379
+FLANNELD_ETCD_CAFILE=/etc/ssl/etcd/ca.crt
+FLANNELD_ETCD_CERTFILE=/etc/ssl/etcd/etcd-client.crt
+FLANNELD_ETCD_KEYFILE=/etc/ssl/etcd/etcd-client.key
+FLANNELD_IFACE=$PRIMARY_HOST_IP
+FLANNELD_PUBLIC_IP=$PRIMARY_HOST_IP
+EOF
+root@kubernetes:/tmp$ cat > /etc/systemd/system/flanneld.service << EOF
+[Unit]
+Description=Network fabric for containers
+Documentation=https://github.com/coreos/flannel
+Requires=networking.service
+Before=docker.service
+After=networking.service
+
+[Service]
+Type=notify
+Restart=always
+RestartSec=5
+EnvironmentFile=/etc/flanneld/options.env
+LimitNOFILE=40000
+LimitNPROC=1048576
+ExecStartPre=/sbin/modprobe ip_tables
+ExecStartPre=/bin/mkdir -p /run/flanneld
+
+ExecStart=/usr/local/bin/flanneld --ip-masq=true
+
+# Update docker options
+ExecStartPost=/opt/flanneld/mk-docker-opts.sh -d /run/flanneld/docker_opts.env -i
+
+[Install]
+WantedBy=multi-user.target
+EOF
+root@kubernetes:/tmp$ systemctl enable flanneld
+root@kubernetes:/tmp$ systemctl start flanneld
+root@kubernetes:/tmp$ rm -rf flannel-0.5.5
+root@kubernetes:/tmp$ rm -f flannel-0.5.5-linux-amd64.tar.gz
+{% endhighlight %}
+
+## Docker engine
+
+As you can see in _/etc/systemd/system/flanneld.service_ the script _mk-docker-opts.sh_ creates environment variables with the file _/run/flanneld/docker_opts.env_.
+These variables are used by the Docker daemon.
+We use the [drop-in feature](https://www.freedesktop.org/software/systemd/man/systemd.unit.html) of *Systemd*.
+So there is no need to change the standard service definition of Docker.
+The [Installation of the Docker engine](https://docs.docker.com/engine/installation/linux/debian/) is done in the standard way.
```

Comment:
The [installation of the Docker engine](https://docs.docker.com/engine/installation/linux/debian/) is done in the standard way.